tests/ArrayKeysTest.php

```diff
@@ -0,0 +1,137 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Psalm\Tests;
+
+use Psalm\Tests\Traits\InvalidCodeAnalysisTestTrait;
+use Psalm\Tests\Traits\ValidCodeAnalysisTestTrait;
+
+class ArrayKeysTest extends TestCase
```

Comment:
Just checked for clarity: all those tests already work on master. Those aren't tests that are solved by the current PR

Reply:
Yes, but they are needed to ensure the functionality still works. While developing, I broke many of these and I think it's nice, if somebody else couldn't break it too 😀

Reply:
yeah, this is useful, I Just try to grasp the extent of the change so knowing what snippets are actually better/different than before is useful for me, so I noted it for other reviewers too